qiita_db/software.py

```diff
@@ -579,10 +579,27 @@ def from_file(cls, fp, update=False):
 
                 if (client_id != instance.client_id or
                         client_secret != instance.client_secret):
-                    raise qdb.exceptions.QiitaDBOperationNotPermittedError(
-                        'The (client_id, client_secret) pair of the plugin '
-                        '"%s" version "%s" does not match the one in the '
-                        'system' % (name, version))
+                    if update:
+                        sql = """INSERT INTO qiita.oauth_identifiers
```

Comment:
I think I don't understand your question? This is inserting if it doesn't already exists.